code/game/objects/items/_item_interactions.dm

```diff
@@ -40,3 +40,76 @@
 
 /decl/interaction_handler/storage_open/invoked(atom/target, mob/user, obj/item/prop)
 	target?.storage?.open(user)
+
+/decl/interaction_handler/wash_hands
+	name = "Wash Hands"
+	expected_target_type = /atom
+	interaction_flags = INTERACTION_NEEDS_PHYSICAL_INTERACTION
+
+/decl/interaction_handler/wash_hands/is_possible(atom/target, mob/user, obj/item/prop)
+	. = ..() && target?.reagents?.has_reagent(/decl/material/liquid/water, 150)
+	if(.)
+		for(var/hand_slot in global.all_washable_hand_slots)
+			if(user.get_organ(hand_slot))
+				return TRUE
+
+/decl/interaction_handler/wash_hands/invoked(atom/target, mob/user, obj/item/prop)
+
+	// Probably needs debounce and do_after() but showers and wading into water don't, so whatever.
+	if(!target?.reagents?.has_reagent(/decl/material/liquid/water, 150)) // To avoid washing your hands in beakers.
+		to_chat(user, SPAN_WARNING("\The [src] doesn't have enough water in it to wash your hands."))
+		return
+
+	var/found_hand = FALSE
+	for(var/hand_slot in global.all_washable_hand_slots)
+		if(user.get_organ(hand_slot))
+			found_hand = TRUE
+			break
+	if(!found_hand)
+		return
+
+	var/decl/pronouns/pronouns = user.get_pronouns()
+	user.visible_message(
+		SPAN_NOTICE("\The [user] washes [pronouns.his] hands in \the [target]."),
```

Comment:
this will, for river turfs, say `Penelope Haze washes her hands in the mud.`, which is admittedly kind of funny but gives a misleading impression. wonder if we could generalize `/turf/proc/get_trench_name()` somehow or if it'd be better to just hardcode a turf check here

Reply:
oh yeah `get_trench_name()` is just for trenches filled with water, and in handwashing it will always be filled with water